allocator.go

```diff
@@ -184,3 +218,66 @@ func (a *Allocator) Allocate() (ok bool, assignments map[RangeID][]NodeID) {
 	}
 	return true, res
 }
+
+// Minimize the difference absolute difference from the previous assignment and the current assignment
+func (a *Allocator) adhereToMinimalChurnConstraint() {
+
+	if !a.opts.churnOptions.WithMinimalChurn && !a.opts.churnOptions.withMaxChurn {
+		return
+	}
+
+	var vars = make([]cpsatsolver.IntVar, 0, 2*len(a.ranges)*len(a.nodes))
+	var coefficients = make([]int64, 0, len(a.ranges)*len(a.nodes))
+	o := toMap(a.opts.churnOptions.prevAssignment)
+
+	for _, r := range a.ranges {
+		for _, n := range a.nodes {
+			termCoefficient := int64(1)
+			if val := o[r.id][n.id]; val == 1 {
+				prevAllocation := a.model.NewIntVar(1, 1, fmt.Sprintf("Previous: r%d-on-n%d", r.id, n.id))
```

Comment:
Why are we creating new intvars here? Why not re-use the literals we already have for the rangeID-nodeID pair? I think we can use literals + sum over literals like so:

```go
if a.Options.DisableMaxChurn { // Limit the number of items that are moved around.
  if len(a.LastPlacement) != 0 {
    var literals []solver.Literal
    for _, bin := range a.Bins {
      for _, item := range a.Items {
        if !a.placed(item, bin) {
          continue
        }

        // This item-bin pair was previously placed. If it's not
        // placed in the same way this round (the literal will be
        // assigned false). Since we want to count it the number of
        // times this occurs, get its negation.
        literals = append(literals, a.get(item, bin).Not())
      }
    }
    model.AddConstraints(
      solver.NewAtMostKConstraint(a.MaxChurn(), literals...), // Ensure that at most K new placements happen.
    )
    model.Minimize(solver.Sum(literals...)) // Minimize the number of new placements.    
  }
}
```